qa/tasks/check_counter.py

```diff
@@ -96,7 +112,9 @@ def end(self):
 
                     if val is not None:
                         log.info(f"Daemon {daemon_type}.{daemon_id} {name}={val}")
-                        if val >= minval:
+                        if isinstance(minval, int) and val >= minval:
+                            seen.add(name)
+                        elif isinstance(expected_val, int) and val == expected_val:
```

Comment:
@kotreshhr This caused the following errors:

```
2023-03-28T10:34:36.302 ERROR:teuthology.run_tasks:Manager failed: check-counter
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_main/teuthology/run_tasks.py", line 192, in run_tasks
    suppress = manager.__exit__(*exc_info)
  File "/home/teuthworker/src/git.ceph.com_teuthology_main/teuthology/task/__init__.py", line 132, in __exit__
    self.end()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_9d12d2cb27d4a45c58218390bfbb56820b010b77/qa/tasks/check_counter.py", line 117, in end
    elif isinstance(expected_val, int) and val == expected_val:
UnboundLocalError: local variable 'expected_val' referenced before assignment
```

More detail please see https://pulpito.ceph.com/yuriw-2023-03-28_00:15:34-fs-wip-yuri5-testing-2023-03-27-1315-distro-default-smithi/.

Could you have a look ? Thanks.

Reply:
Thanks Xiubo. I will take a look.

Reply:
Fixed it. Ran the failed one with the fix. It works. Hopefully this is the last failure.
http://pulpito.front.sepia.ceph.com/khiremat-2023-03-29_06:17:28-fs-wip-yuri5-testing-2023-03-27-1315-distro-default-smithi/

Reply:
@yuriw This needs a qa re-run after this fix.